mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td

```diff
@@ -381,13 +383,14 @@ def LLVM_LoadOp : LLVM_MemAccessOpBase<"load",
     $res = $_builder.create<LLVM::LoadOp>($_location, $_resultType, $addr,
         alignment, loadInst->isVolatile(),
         loadInst->hasMetadata(llvm::LLVMContext::MD_nontemporal),
+        loadInst->hasMetadata(llvm::LLVMContext::MD_invariant_load),
         convertAtomicOrderingFromLLVM(loadInst->getOrdering()),
         getLLVMSyncScope(loadInst));
   }];
   let builders = [
     OpBuilder<(ins "Type":$type, "Value":$addr,
       CArg<"unsigned", "0">:$alignment, CArg<"bool", "false">:$isVolatile,
-      CArg<"bool", "false">:$isNonTemporal,
+      CArg<"bool", "false">:$isNonTemporal, CArg<"bool", "false">:$isInvariantLoad,
```

Comment:
ultra nit: I would probably go with `isInvariant` since the flag is already on the load op?

Reply:
Yeah, fair, though also `invariant.load` is the metadata over there

Reply:
Ah I see. I guess for metadata the long name makes a bit more sense since it may collide with other metadata kinds. Up to you what you prefer I guess.

Reply:
I went with your suggestion